lib/xmain/opts.go

```diff
@@ -0,0 +1,132 @@
+package xmain
+
+import (
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/pflag"
+	"oss.terrastruct.com/cmdlog"
+	"oss.terrastruct.com/xos"
+)
+
+type Opts struct {
+	args  []string
+	flags *pflag.FlagSet
+	env   *xos.Env
+	log   *cmdlog.Logger
+
+	registeredEnvs []string
+}
+
+func NewOpts(env *xos.Env, args []string, log *cmdlog.Logger) *Opts {
+	flags := pflag.NewFlagSet("", pflag.ContinueOnError)
+	flags.SortFlags = false
+	flags.Usage = func() {}
+	flags.SetOutput(io.Discard)
+	return &Opts{
+		args:  args,
+		flags: flags,
+		env:   env,
+		log:   log,
+	}
+}
+
+func (o *Opts) Help() string {
+	b := &strings.Builder{}
+	o.flags.SetOutput(b)
+	o.flags.PrintDefaults()
+
+	if len(o.registeredEnvs) > 0 {
+		b.WriteString("\nYou may persistently set the following as environment variables (flags take precedent):\n")
```

Comment:
This isn't ideal. I wanted our own PrintDefaults that includes the env that you can set on each flag's help so that you don't have to go back and forth in the help between the env variable and which flag it sets and the docs for that flag.

Reply:
yeah i understand. `pflag` doesn't expose that extensibly.

Reply:
Yes it does? https://pkg.go.dev/github.com/spf13/pflag#Flag

Reply:
PrintDefaults ends up here: https://github.com/spf13/pflag/blob/master/flag.go#L684

that's a fairly large chunk of code to copy paste. what i meant was that they don't let this part be extensible, like if i wanted to add some custom msg to each flag line.

am i understanding u correctly that this is what u want to override?

Reply:
Yea copy and paste, that code hasn't changed in millenia I guarantee it and we'll want to integrate our config file field names too later #108

Reply:
Add a comment ofc for where the code was copied ffrom.